lib/io/buffered/reader.fz

```diff
@@ -0,0 +1,223 @@
+# This file is part of the Fuzion language implementation.
+#
+# The Fuzion language implementation is free software: you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as published
+# by the Free Software Foundation, version 3 of the License.
+#
+# The Fuzion language implementation is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+# License for more details.
+#
+# You should have received a copy of the GNU General Public License along with The
+# Fuzion language implementation.  If not, see <https://www.gnu.org/licenses/>.
+
+
+# -----------------------------------------------------------------------
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion standard library feature io.buffered.reader
+#
+# -----------------------------------------------------------------------
+
+
+# buffered.reader effect allows buffered reading
+# by using the given Read_Provider
+#
+# note: anything in the buffer when effect is uninstalled will be discarded.
+#
+reader(private rp Read_Provider, buf_size i32, buffer array u8 | io.end_of_file) : try is
```

Comment:
I do not understand why this inherits from `try`. There is one place where `raise` is called, but two other places where `try.env` is used. So it looks to me as if two different types of `try` effects are mixed. Please explain. 

Without inheriting the try effect, my impression is that a buffered reader does not need to be an effect at all. But it does manage a buffer, so it contains mutable fields and a mutable array, so this should IMHO use a type parameter `LM type : mutate` to perform these mutations in a way that allows a local instance of mutate without adding one more effect.